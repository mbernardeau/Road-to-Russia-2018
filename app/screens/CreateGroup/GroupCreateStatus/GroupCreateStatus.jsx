import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import Snackbar from '@material-ui/core/Snackbar'
import IconButton from '@material-ui/core/IconButton'
import CloseIcon from '@material-ui/icons/Close'

import './GroupCreateStatus.scss'

const GroupCreateStatus = ({ status, group: { name, joinKey }, reason, handleClose }) => (
  <Snackbar
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'left',
    }}
    open={!!status}
    autoHideDuration={6000}
    onClose={handleClose}
    SnackbarContentProps={{
      'aria-describedby': 'message-id',
    }}
    message={
      <span id="message-id">
        {status === 'failed' ? (
          reason
        ) : (
          <Fragment>
            Tribu {name} créée avec la clé <b>{joinKey}</b> &nbsp;<Link
              className="group-create-link"
              to="/admingroups"
            >
              En voir plus
            </Link>
          </Fragment>
        )}
      </span>
    }
    action={[
      <IconButton key="close" aria-label="Close" color="inherit" onClick={handleClose}>
        <CloseIcon />
      </IconButton>,
    ]}
  />
)

GroupCreateStatus.defaultProps = {
  group: {},
}

GroupCreateStatus.propTypes = {
  status: PropTypes.string,
  group: PropTypes.shape({
    name: PropTypes.string,
    joinKey: PropTypes.string,
  }),
  reason: PropTypes.string,
  handleClose: PropTypes.func.isRequired,
}

export default GroupCreateStatus
