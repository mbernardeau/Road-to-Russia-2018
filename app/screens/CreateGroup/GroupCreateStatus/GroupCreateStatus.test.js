import React from 'react'
import renderer from 'react-test-renderer'

import GroupCreateStatus from './GroupCreateStatus'

jest.mock('material-ui/Snackbar', () => 'Snackbar')
jest.mock('material-ui/IconButton', () => 'IconButton')
jest.mock('@material-ui/icons/Close', () => 'CloseIcon')

describe('screens/CreateGroup/GroupCreateStatus', () => {
  it('renders correctly without props', () => {
    const props = {
      handleClose: jest.fn(),
    }

    const tree = renderer.create(<GroupCreateStatus {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly when failed props', () => {
    const props = {
      handleClose: jest.fn(),
      status: 'failed',
      reason: 'Nos√©',
    }

    const tree = renderer.create(<GroupCreateStatus {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly when success', () => {
    const props = {
      handleClose: jest.fn(),
      status: 'success',
      group: {},
    }

    const tree = renderer.create(<GroupCreateStatus {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
