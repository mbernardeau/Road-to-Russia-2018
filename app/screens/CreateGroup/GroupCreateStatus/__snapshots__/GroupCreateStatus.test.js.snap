// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/CreateGroup/GroupCreateStatus renders correctly when failed props 1`] = `
<Snackbar
  SnackbarContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  action={
    Array [
      <IconButton
        aria-label="Close"
        color="inherit"
        onClick={[MockFunction]}
      >
        <CloseIcon />
      </IconButton>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  message={
    <span
      id="message-id"
    >
      Nosé
    </span>
  }
  onClose={[MockFunction]}
  open={true}
/>
`;

exports[`screens/CreateGroup/GroupCreateStatus renders correctly when success 1`] = `
<Snackbar
  SnackbarContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  action={
    Array [
      <IconButton
        aria-label="Close"
        color="inherit"
        onClick={[MockFunction]}
      >
        <CloseIcon />
      </IconButton>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  message={
    <span
      id="message-id"
    >
      Demande envoyée pour la tribu undefined
    </span>
  }
  onClose={[MockFunction]}
  open={true}
/>
`;

exports[`screens/CreateGroup/GroupCreateStatus renders correctly without props 1`] = `
<Snackbar
  SnackbarContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  action={
    Array [
      <IconButton
        aria-label="Close"
        color="inherit"
        onClick={[MockFunction]}
      >
        <CloseIcon />
      </IconButton>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  message={
    <span
      id="message-id"
    >
      Demande envoyée pour la tribu undefined
    </span>
  }
  onClose={[MockFunction]}
  open={false}
/>
`;
